src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -54,24 +54,58 @@ public async Task<ImmutableArray<CodeAction>> AddConstructorParametersFromMember
             }
         }
 
+        /// <summary>
+        /// Try to find a constructor in <paramref name="containingType"/> whose parameters is the subset of <paramref name="parameters"/> by comparing name.
+        /// If multiple constructors meet the condition, the one with more parameters will be returned.
+        /// It will not consider those constructors as potential candidates
+        /// 1. Constructor with empty parameter list.
+        /// 2. Constructor's parameter list contains 'ref' or 'params'
+        /// </summary>
+        protected override IMethodSymbol GetDelegatedConstructor(
+            INamedTypeSymbol containingType,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+            return containingType.InstanceConstructors
+                .WhereAsArray(constructor => IsParamtersContainedInConstructor(constructor, parameters.SelectAsArray(p => p.Name)))
+                .OrderByDescending(constructor => constructor.Parameters.Length)
+                .FirstOrDefault();
+        }
+
+        private bool IsParamtersContainedInConstructor(
+            IMethodSymbol constructor,
+            ImmutableArray<string> parametersName)
+        {
+            var constructorParams = constructor.Parameters;
+            if (constructorParams.Length > 0
+                && constructorParams.All(parameter => parameter.RefKind == RefKind.None)
+                && !constructorParams.Any(p => p.IsParams))
+            {
+                return parametersName.Except(constructorParams.Select(p => p.Name)).Any();
+            }
+            else
+            {
+                return false;
+            }
+        }
```

Comment:
can be simplified to `return consturctorParams.Length > 0 && ... && parameterNames.Except...`